zebrad/tests/common/checkpoints.rs

```diff
@@ -504,7 +505,8 @@ pub fn wait_for_zebra_checkpoints_generation<
 /// Returns an approximate `zebrad` tip height, using JSON-RPC.
 #[tracing::instrument]
 pub async fn zebrad_tip_height(zebra_rpc_address: SocketAddr) -> Result<Height> {
-    let client = RpcRequestClient::new(zebra_rpc_address);
+    let auth_cookie = cookie::get(RpcConfig::default().cookie_dir).expect("cookie should exist");
+    let client = RpcRequestClient::new(zebra_rpc_address, auth_cookie);
```

Comment:
We write the cookie to the disk, and then read it even in tests that are not related to the authentication. We could generate the cookie for tests without writing it to the disk at all, and instantiate it only through `lazy_static!`.

Reply:
https://github.com/ZcashFoundation/zebra/pull/8900#discussion_r1798481844